Automattic/wp-calypso
# Framework: Upgrade React to 15.3.0

This pull request seeks to bump the minor version of React to its latest version, from 15.1.0 to 15.3.0.

- [15.2.0 changelog](https://github.com/facebook/react/blob/master/CHANGELOG.md#1520-july-1-2016)
- [15.3.0 changelog](https://github.com/facebook/react/blob/master/CHANGELOG.md#1530-july-29-2016)

While there are a number of general fixes, I'm most allured by the promise for more _speeeeed_ (https://github.com/facebook/react/pull/7005). Well, potential speed anyways.

__Testing instructions:__

Verify that there are no regressions in using the application from the upgrade. Examine changes included in the upgrade changelog and test against suspected use-cases that may be affected.

Your console will become very noisy as a result of this particular change:

>Add warning for unknown properties on DOM elements.

I imagine many of these warnings are a result of applying a blanket spread of `this.props` to a DOM element. It's probably a good thing that we seek to clean these up, as I might imagine they have the potential to lead to some unexpected circumstances. In any case, it is what it is. ¯\\\_(ツ)\_/¯

Test live: https://calypso.live/?branch=update/react-15-2
